packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -98,59 +188,122 @@ function ShoppingPane(props) {
     insertElement: state.actions.insertElement,
   }));
 
-  const insertProduct = useCallback(() => {
-    insertElement('product', {
-      width: 25,
-      height: 25,
-      product,
-    });
-    setProduct(null);
-  }, [insertElement, product]);
+  const deleteProduct = useCallback(
+    (product) => {
+      const element = pageProducts.find(
+        (item) => item.productId === product.productId
+      );
+      if (element) {
+        deleteElementById({ elementId: element.elementId });
+      }
+    },
+    [deleteElementById, pageProducts]
+  );
+
+  const insertProduct = useCallback(
+    (product) => {
+      insertElement('product', {
+        width: 25,
+        height: 25,
+        product,
+      });
+    },
+    [insertElement]
+  );
 
   if (!isEnabled) {
     return null;
   }
 
-  const dropDownParams = {
-    hasSearch: true,
-    lightMode: true,
-    onChange,
-    getOptionsByQuery: getProductsByQuery,
-    selectedId: product?.productId,
-    dropDownLabel: __('Product', 'web-stories'),
-    placeholder: isLoading ? __('Loading…', 'web-stories') : '',
-    disabled: isLoading ? true : isSaving,
-    primaryOptions: isLoading ? [] : initialOptions,
-    zIndex: 10,
-  };
+  const RenderItemOverride = forwardRef(
+    ({ option, isSelected, ...rest }, ref) => {
+      const onPage = pageProducts.find(
+        (item) => item.productId === option?.product.productId
+      );
+      const activeIcon = (
+        <>
+          <Icons.ProductCheck className="check" />{' '}
+          <Icons.ProductRemove className="remove" />
+        </>
+      );
+
+      const handleClick = onPage
+        ? () => {
+            deleteProduct(option?.product);
+          }
+        : () => {
+            insertProduct(option?.product);
+          };
+      const src = option?.product?.productImages[0]?.url || '';
+      return (
+        // eslint-disable-next-line styled-components-a11y/click-events-have-key-events -- click events in the contained buttons.
+        <StyledListItem
+          ref={ref}
+          width={option.width}
+          active={isSelected}
+          {...rest}
+          onClick={null}
+        >
+          <StyledProductContainer>
```

Comment:
For later, this whole product list item component would be a good candidate to extract into its own file and create a storybook story for.

Reply:
Broke out the Product list and added a story